pkg/cli/modes/histwalk.go

```diff
@@ -120,3 +122,15 @@ func (w *histwalk) updatePending() {
 		}
 	})
 }
+
+func (w *histwalk) Accept() error {
```

Comment:
- The state manipulation _removes_ the prefix, that seems wrong - if I type `echo`, press Up, get `echo foo`, and call this, the buffer will be ` foo`.

- The correct behavior is implemented by `(*tk.CodeAreaState).ApplyPending`, which is used in the `Handle` method on line 62.

- For consistency with `edit:listing:accept`, this should also terminate histwalk mode.